### ISSUE TITLE
empty file on video recording, 8.7.1 macOS

### ISSUE THREAD

**Issue by schlomo** [webcamoid/webcamoid#297]
## Summary

Recording a video creates an empty file

## Current Behavior

When recording a video from the webcam the output file is empty, no matter which webcam or video codec setting I use. I tried MJPEG and YUV2 in different resolutions and different codecs/formats like iPod H.264, DV etc.

The console log looks like this:

```
$ /Applications/webcamoid.app/Contents/MacOS/webcamoid
WARNING: Cannot find style "Universal" - fallback: "/Applications/webcamoid.app/Contents/Resources/qml/QtQuick/Controls/Styles/Desktop"
qrc:/Webcamoid/share/qml/RecordConfig.qml:126:5: QML Label: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
qrc:/VideoCapture/share/qml/main.qml:276: TypeError: Cannot call method 'createCameraControls' of null
qrc:/VirtualCamera/share/qml/main.qml:25:1: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
qrc:/Webcamoid/share/qml/RecordConfig.qml:126:5: QML Label: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
qrc:/Webcamoid/share/qml/RecordConfig.qml:126:5: QML Label: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
qrc:/Webcamoid/share/qml/RecordConfig.qml:126:5: QML Label: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
qrc:/Webcamoid/share/qml/RecordConfig.qml:126:5: QML Label: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
qrc:/Webcamoid/share/qml/RecordConfig.qml:126:5: QML Label: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
Can't write header:  Error number 22 occurred
```

Taking photos and the preview both function.

With Quicktime or other tools video recording from the webcam is no problem.

## Expected Behavior

Create a video file with content. #53 seems to be similar but on Gentoo and fixed several years ago.

## Your Environment

* Webcamoid information: 8.7.1
* Operating System information: macOS 10.15.4 Catalina x86_64
* Any other useful information: exactly same behavior with the development version
* Webcam is Logitech C920

**Comment by hipersayanX** [comment]
Which file format, with which video codec, with which audio codec, gives you the ```Can't write header:  Error number 22 occurred``` error message?
Give me a bounded set of options so I can test this bug quickly.

**Comment by schlomo** [comment]
![image](https://user-images.githubusercontent.com/101384/81103927-08eaa480-8f12-11ea-98a5-6029670f3426.png)
I didn't configure anything in detail.

Is there a way to write a detailed debug log?

**Comment by hipersayanX** [comment]
Everything seems to be correct there, also thanks for testing the daily build. Will check.

>Is there a way to write a detailed debug log?

Webcamoid logging is pretty much rudimentary, what you see in the terminal is what you get, and it could be incomplete. The only way of fixing this kind of bugs is getting into the source code and tracing it until you discover where is the problem.

**Comment by schlomo** [comment]
We can do a 1:1 video call if you want to look at my system

**Comment by hipersayanX** [comment]
Sorry but I don't give private support.

**Comment by schlomo** [comment]
Wasn't asking for support, I was offering access to the problem if you want. No worries.

**Comment by hipersayanX** [comment]
Problem solved in 9.0.0, closing.
